yt_dlp/extractor/cda.py

```diff
@@ -153,6 +152,10 @@ def _api_extract(self, video_id):
 
         self._sort_formats(formats)
 
+        if meta.get('premium') and not meta.get('premium_free') and len(formats) == 0:
+            raise ExtractorError(
+                'No video formats and video requires CDA Premium - do you have a subscription?', expected=True)
```

Comment:
```suggestion
        if meta.get('premium') and not meta.get('premium_free') and not formats:
            self.raise_login_required('Requires CDA Premium')
```

Reply:
this code is running if the user has provided valid credentials already, so I don't think `raise_login_required` is appropriate?